### ISSUE TITLE
TypeORM 0.3+ Not re-using mysql connections

### ISSUE THREAD

**Issue by davecarlson** [typeorm/typeorm#8987]
## Issue Description
Existing Connection to RDS is not reused. On every API invocation, a new connection is created

### Expected Behavior
Like the 0.2 line, I would expect to be able to re-use connections:
```
const connectToDatabase = async (name = 'default') => {
  const AppDataSource = new TypeORM.DataSource(options[name]);
  // This is always false, so no reuse
  if (!AppDataSource.isInitialized) {
    await AppDataSource.initialize() 
  }
  return AppDataSource
}
```

### Actual Behavior
<img width="441" alt="Screenshot 2022-05-09 at 6 50 06 pm" src="https://user-images.githubusercontent.com/299702/167468700-df8150fd-38a3-4c00-8b91-9d7411708bdd.png">

```
// include the output in code tags like these!
```

### Steps to Reproduce

1.
2.

```typescript
default: {
    type: 'mysql',
    host: process.env.DB_HOST,
    port: 3306,
    username: process.env.DB_USER,
    password: process.env.DB_PASS,
    database: process.env.DB_NAME,
    entities: [
    ...entities
    ],
    synchronize: false,
    logging: false,
    timezone: 'Z'
  },
  sync: {}
```

### My Environment

Node v16
Next.js v2
Amazon RDS (mySQL 8)

| Dependency          | Version  |
| ---                 | ---      |
| Operating System    |          |
| Node.js version     | x.y.zzz  | 
| Typescript version  | x.y.zzz  | 
| TypeORM version     | x.y.zzz  | 

### Relevant Database Driver(s)

| DB Type           | Reproducible  |
|-------------------| ---           |
| `aurora-mysql`    | no            |
| `aurora-postgres` | no            |
| `better-sqlite3`  | no            |
| `cockroachdb`     | no            |
| `cordova`         | no            |
| `expo`            | no            |
| `mongodb`         | no            |
| `mysql`           | yes            |
| `nativescript`    | no            |
| `oracle`          | no            |
| `postgres`        | no            |
| `react-native`    | no            |
| `sap`             | no            |
| `spanner`         | no            |
| `sqlite`          | no            |
| `sqlite-abstract` | no            |
| `sqljs`           | no            |
| `sqlserver`       | no            |

### Are you willing to resolve this issue by submitting a Pull Request?

 - ✅ Yes, I have the time, and I know how to start.
 - ✖️ Yes, I have the time, but I don't know how to start. I would need guidance.
 - ✖️ No, I don’t have the time, but I can support (using donations) development.
 - ✖️ No, I don’t have the time and I’m okay to wait for the community / maintainers to resolve this issue.

**Comment by davecarlson** [comment]
See last 24 hours on our rds instance connections count. We pushed the change from 0.2 line to 0.3 today, and you can clearly see where the issue starts, and where we backed it out
<img width="893" alt="Screenshot 2022-05-09 at 7 39 18 pm" src="https://user-images.githubusercontent.com/299702/167475823-75c4f7a0-c66b-4219-96c1-dd3767d77792.png">
!

**Comment by muhammadsalmanafzal** [comment]
Same. We are using mySql and would really like the previous implementation of creating connections and getting repository and and then use its functions, back. Because this way, we always have to get repository from the the new connection made and not from the single connection that was established in previous version (0.2). It also breaks the scalability in a sense since we then always have to use the connection from DataSource for the 10's of repository functions made.

**Comment by davecarlson** [comment]
I found a "solution" to this, but not a nice one. You also lose the option of passing in a named connection

If you move the creation out of the function, it does it less often, as node is caching the import:
```
const AppDataSource = new TypeORM.DataSource(options['default]);
const connectToDatabase = async () => {
  
  // This is always false, so no reuse
  if (!AppDataSource.isInitialized) {
    await AppDataSource.initialize() 
  }
  return AppDataSource
}
```

**Comment by ailsoncgt** [comment]
Any solution here? I'm facing same problem ...

**Comment by davecarlson** [comment]
> 

Long term we just moved to prisma! > 100 api routes, took ~ 4 months. however, we also noticed that our average CPU usage went from 40-50% to ~ 15%. So much that we actually dropped down an instance level on AWS and it's STILL faster

**Comment by waggyman** [comment]
I agree, I believe it will be really helpful to use previous connection

**Comment by shazadmaved** [comment]
Any easy work around on this?
